MainRepository/Program.cs

```diff
@@ -1,7 +1,39 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
+﻿TQueue<string> t = new ();
+t.EnQueue ("Hello");
+t.EnQueue ("Hi");
+t.EnQueue ("Good");
+t.EnQueue ("Wow");
+Console.WriteLine ("Removing " + t.DeQueue ());
+t.EnQueue ("Super");
+
+/// <summary>Implementation of Circular Queue class using array</summary>
+class TQueue<T> {
+   /// <summary>Adds element in the queue</summary>
+   public void EnQueue (T a) {
+      if (((mRear + 1) % mQueueArray.Length) == mFront) throw new Exception ("Overflow");
+      if (mFront == -1) mFront = 0;
+      mRear = (mRear + 1) % mQueueArray.Length;
+      mQueueArray[mRear] = a;
+      Console.WriteLine ($"Inserting {a}");
+   }
+
+   /// <summary>Removes element in the queue</summary>
+   public T DeQueue () {
+      if (IsEmpty () == true) throw new Exception ("Underflow");

```

Comment:
IsEmpty returns bool no need to compare it with truee
```suggestion
      if (IsEmpty ()) throw new Exception ("Underflow");

```